### PR TITLE
Multiple projects

### PR DESCRIPTION
**Pull request by dododedodonl**
This pull requests adds code to support multiple firebase projects in laravel via a new facade.

Since there is a breaking change in the configuration file, a major version bump is needed. Also, all existing facades are deprecated in favor of a single new facade to support multiple projects. They are however included for an easier upgrade path.

### CODE REVIEW

File: src/FirebaseProject.php
Comment (jeromegamez) [inline src/FirebaseProject.php:11]: I usually don't include FQNs in PHPDocs, so this "risks" being changed to `Kreait\Firebase\Factory` (due to the existing import) or to `Factory` with an additional `use` statement somewhen in the future 🙈  (similarly elsewhere)

File: src/FirebaseProject.php
Comment (jeromegamez) [inline src/FirebaseProject.php:18]:
```suggestion
    protected $auth;
```

`null` is already the default value - it of course doesn't matter in the functionality, but while I usually like being explicit, class properties that are null by default are one exception 🙈 (even if you'll certainly find occurrences where I don't follow my own rules 😂)

File: src/FirebaseProject.php
Comment (jeromegamez) [inline src/FirebaseProject.php:46]:
```suggestion
        if (!$this->auth) {
```
Again, this is just style, but this is something I would otherwise change in a future commit, or as soon as I introduce PHP CS Fixer (with similar rules as in `kreait/firebase-php`)

File: src/FirebaseProjectManager.php
Comment (jeromegamez) [inline src/FirebaseProjectManager.php:13]: Could you check if this equally works with `Illuminate\Contracts\Container\Container`? This would probably mean that getting the config would have to be implemented differently, but I remember changing `Application` to `Container` in the past for better Lumen support.

File: src/ServiceProvider.php
Comment (jeromegamez) [inline src/ServiceProvider.php:8]: Same as in `FirebaseProjectManager` - I used `Container` in other parts of the app because of Lumen and if it's possible, we should stick with that

**Review by jeromegamez — CHANGES_REQUESTED** [review]
That's a great PR, thank you so much! Most of my change requests are not concerning the new behavior, just my personal preference as a maintainer, I hope you can see past that 😅 

I'll try to test this for real soon, but this looks really, really good!

**Comment by jeromegamez** [comment]
Could you please also remove support for Laravel/Illuminate 5.8 in the composer.json and GitHub actions? I think `orchestra/testbench ^3.8.6` can then go as well.

File: README.md
Comment (dododedodonl) [reply README.md:13]: I had a plugin enabled that auto generated a table of contents, did not notice it

File: src/FirebaseProject.php
Comment (dododedodonl) [reply src/FirebaseProject.php]: Forgot to remove it when I removed the property types :)

File: src/FirebaseProjectManager.php
Comment (dododedodonl) [reply src/FirebaseProjectManager.php:13]: I'll have a look into it. I changed it to copy the `resolveCredentials` method to `FirebaseProjectManager`, as `basePath` does not exist on `Container`.

File: src/FirebaseProjectManager.php
Comment (dododedodonl) [reply src/FirebaseProjectManager.php:17]: Missed that one, thanks for finding it :)

**Comment by dododedodonl** [comment]
I think I incorporated all your suggestions :smiley: 
 
* Added docblock suggestion for facades
* Added php-cs-fixer (copied config from kreait/firebase-php, commented out 2 rules)
* Moved upgrade guide to UPGRADE.md
* Removed Laravel and Lumen 5.8 support, removed orchestra/testbench:^3.6.8, lowered minimums to 4.0 and 5.0
* Changed `Illuminate\Contracts\Foundation\Application` to `Illuminate\Contracts\Container\Container`
* Other small fixes

**Comment by jeromegamez** [comment]
Sorry for the late response (again) - I just tested it locally and it works just fine, and using the new Facade is so much more enjoyable!

I will merge this now, and after integrating the other open PR and making some minor changes, I'll release this as a new version! This will be a nice new major release, thank you so much! 🌺

**Pull request merged**
